src/H5FDcore.c

```diff
@@ -407,6 +414,32 @@ H5FD__core_write_to_bstore(H5FD_core_t *file, haddr_t addr, size_t size)
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5FD__core_write_to_bstore() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__core_get_default_config
+ *
+ * Purpose:     Retrieves a default configuration for this VFD when no
+ *              configuration information has been provided.
+ *
+ * Return:      Valid Core VFD configuration information pointer (can't
+ *                  fail)
+ *
+ *-------------------------------------------------------------------------
+ */
+static inline const H5FD_core_fapl_t *
+H5FD__core_get_default_config(void)
+{
+    char *driver = HDgetenv("HDF5_DRIVER");
```

Comment:
I feel like it would be better to use a macro instead of "HDF5_DRIVER" everywhere. Just in case that environment variable has to be renamed one day or someone messes up the environment variable name, this way the compiler would be able to detect it for us.

Reply:
Added a few macros to H5public.h for this and some other related environment variables.